pavelib/assets.py

```diff
@@ -172,14 +195,23 @@ def compile_templated_sass(systems, settings):
     """
     for sys in systems:
         sh(django_cmd(sys, settings, 'preprocess_assets'))
+        print("\t\tFinished preprocessing assets.")
 
 
-def process_xmodule_assets():
+def process_xmodule_assets(restart_django_server=False):
     """
     Process XModule static assets.
     """
     sh('xmodule_assets common/static/xmodule')
 
+    if restart_django_server:
+        # The python server needs to be reloaded (for some reason) in order to
```

Comment:
@feanil I scanned the code quickly and I found [this snippet] (https://github.com/edx/edx-platform/blob/master/common/lib/xmodule/xmodule/static_content.py#L176):

```
    for filename, file_content in contents.iteritems():
        output_file = output_root / filename

        not_file = not output_file.isfile()

        # not_file is included to short-circuit this check, because
        # read_md5 depends on the file already existing
        write_file = not_file or output_file.read_md5() != hashlib.md5(file_content).digest()  # pylint: disable=too-many-function-args
        if write_file:
            LOG.debug("Writing %s", output_file)
            output_file.write_bytes(file_content)
        else:
            LOG.debug("%s unchanged, skipping", output_file)
```

I *guess* that this snippet is why the Django server (more specifically the xmodule) is unable to serve the file after it changes.